### ISSUE TITLE
acme: unable to retrieve self-hosted domain validation token

### ISSUE THREAD

**Issue by botter-nidnul** [urbit/urbit#3101]
**Describe the bug**

I moved ~botter-nidnul to a new server (with new IP address) and breached.  Setcap was used to allow urbit to bind to ports below 1024, and http runs on port 80 (`http: live (insecure, public) on 80`)

After `-dns-address &dns-address [%if .178.170.10.5]` was processed and I confirmed that botter-nidnul.arvo.network was pointing to the correct address I went to setup https with acme using:

`:acme &path /network/arvo/botter-nidnul` which output:
```
[%key iter=0 width=2.047]
[%key iter=1 width=2.047]
[%key iter=0 width=2.047]
> :acme &path /network/arvo/botter-nidnul
>=
acme: requesting an https certificate for botter-nidnul.arvo.network
```
then silence, with no announcement of https running on port 443.

I restarted urbit, and it showed only
```
http: live (insecure, public) on 80
http: live (insecure, loopback) on 12321
```

so I then ran `:acme %retry`:
```
> :acme %retry
>=
acme: requesting an https certificate for botter-nidnul.arvo.network
acme: unable to retrieve self-hosted domain validation token via botter-nidnul.arvo.network. please confirm your urbit has network connectivity.
[ p=500
    q
  ~[
    [p='date' q='Tue, 07 Jul 2020 18:25:47 GMT']
    [p='connection' q='keep-alive']
    [p='content-length' q='250']
    [p='server' q='urbit/vere-0.10.7']
    [p='content-type' q='text/html']
  ]
    r
  [ ~
    [ p=250
        q
      \/'<html><head><title>500 Internal Server Error</title></head><body><h1>\/
        Internal Server Error</h1><p>There was an error while handling the req
        uest for &quot;/.well-known/acme-challenge/{is this string sensitive? I removed it}&quot;.</p></body></html>'
      \/                                                                      \/
    ]
  ]
]
acme: retrying certificate request in ~s32..00a1
error in %acme
%arvo-response
! /app/acme/hoon:<[750 44].[750 46]>
! %no-active-order
! /app/acme/hoon:<[750 24].[750 47]>
! /app/acme/hoon:<[750 5].[765 39]>
! /app/acme/hoon:<[749 5].[765 39]>
! %test-trial-effect-fail
! /app/acme/hoon:<[748 5].[765 39]>
! /app/acme/hoon:<[747 5].[765 39]>
! /app/acme/hoon:<[1.195 24].[1.195 38]>
! /app/acme/hoon:<[1.180 5].[1.197 7]>
! /app/acme/hoon:<[1.179 5].[1.197 7]>
! /app/acme/hoon:<[1.178 5].[1.197 7]>
! /app/acme/hoon:<[1.177 5].[1.197 7]>
! /app/acme/hoon:<[1.176 5].[1.197 7]>
! /app/acme/hoon:<[1.175 5].[1.197 7]>
! /app/acme/hoon:<[1.174 5].[1.197 7]>
! /app/acme/hoon:<[1.266 8].[1.266 27]>
! /app/acme/hoon:<[1.266 3].[1.266 27]>
! /app/acme/hoon:<[1.265 3].[1.266 27]>
! /app/acme/hoon:<[1.262 3].[1.266 27]>
! /app/acme/hoon:<[1.261 3].[1.266 27]>
! /app/acme/hoon:<[380 27].[380 54]>
! /app/acme/hoon:<[378 9].[382 11]>
! /app/acme/hoon:<[376 7].[383 9]>
! /app/acme/hoon:<[375 5].[384 17]>
! /app/acme/hoon:<[374 5].[384 17]>
```
it retries at increasing intervals but continues to repeat the `500 Internal Server Error`

**To Reproduce**
Steps to reproduce the behaviour:
1.  `:acme &path /network/arvo/sampel-sipnym`
2.  after the previous fails to start https on port 443 `:acme %retry`

**Expected behaviour**
Let's Encrypt certificate request to complete successfully with eyre then starting up https on port 443

**System (please supply the following information, if relevant):**
 - OS: Raspbian Linux
 - urbit-v0.10.7-aarch64
 - [%base-hash 0vc.5mb6v.nqu78.n190s.qdfcr.5o28s.kcf0q.nuem2.9d4p8.06sp7.dhk61]

**Notify maintainers**
@joemfb

**Comment by botter-nidnul** [comment]
Another user reported running into the same problem on urbit-help, without doing the `:acme &path` command between `-dns-address &dns-address [%if .1.2.3.4]` and `:acme %retry`

**Comment by Fang-** [comment]
Does your home desk hash match the "base hash"? Alternatively, if home doesn't match, but kids does, are `.^(@uv %cz /=home=/sys)` and `.^(@uv %cz /=kids=/sys)` equivalent?

Someone else reported this to me about a week ago. I thought that https://github.com/urbit/urbit/issues/3089#issuecomment-653355919, which has since gone onto livenet, might have fixed this, so you want to make sure you have that. If hashes seem all good, running `|reset` could be worth a try, just in case.

**Comment by botter-nidnul** [comment]
You were correct, /sys hashes don't match:

```
> .^(@uv %cz /=home=/sys)
0v5.rt4gg.48p64.k857b.linm3.03v76.ks2pf.smnlr.lsen8.4o6o5.mgjms
```
```
> .^(@uv %cz /=kids=/sys)
0vi.5c4qf.kj1nm.n1mmi.fpie2.1esj1.a1hkm.7mphs.odudt.djqf9.v2ouo
```

[%base-hash 0vc.5mb6v.nqu78.n190s.qdfcr.5o28s.kcf0q.nuem2.9d4p8.06sp7.dhk61]
[%home-hash 0vj.rea4s.613h7.iu7sr.vj4s3.84jhu.vkk0n.s4chk.v1cuf.46er0.tfcsa] (I have publish notebooks)
[%kids-hash 0vc.5mb6v.nqu78.n190s.qdfcr.5o28s.kcf0q.nuem2.9d4p8.06sp7.dhk61]

I've tried |merge %home our %kids with %fine %meet %mate and %meld strategies. Is there a way to preserve the publish notebooks and merge the %kids desk from within urbit, or do I need to `git clone` the urbit-os-v1.0.28 tag, copy my notebooks into their proper place, then `|commit %home`?

**Comment by belisarius222** [comment]
Ford Fusion broke `|reset`, and I haven't fixed it yet, so don't run that just yet.

You can check whether you at least got the Ford Fusion OTA (which was two or three OTAs ago) by running:
```
=f -build-file /lib/sole/hoon
```
If that produces no errors, you've received that OTA.

**Comment by botter-nidnul** [comment]
I breached just after urbit-os-v1.0.25 was released and booted up with a Ford Fusion-having pill, so at least that much is good.

I'm interested in how to |merge to %home desks that don't want to be merged, without destroying things like publish notebooks in %home. Maybe that doesn't matter? I'm not certain what happens when the notebook files are removed from the desk, i.e. if the notebook is also removed from publish's state and unpublished.

I guess I could just boot a copy of my planet with -L and find out...

**Comment by botter-nidnul** [comment]
Yes, removing the notebook files from the %home desk with a `gem =%that` merge removes the notebook from the publish app.

Seems I was stuck on the 1.0.25 version from the boot pill and later OTAs were being downloaded but not merged. (I had noticed the %kids desk wasn't updating, I had to do `gem =%that` mergers from my sponsor to get it updated)

Immediately after the upgrade to 1.0.28:
```
acme: received https certificate for botter-nidnul.arvo.network
http: restarting servers to apply configuration
http: live (secure, public) on 443
```

I'm still not able to automerge %kids with my updated %home desk though. I'm not certain why, I don't think I did anything unusual with it - just mounted %home and copied a notebook into it.

**Comment by philipcmonk** [comment]
So you were able to merge `%that` from parent's %kids to your %kids?  If you merge from %kids to %home, what error does it give?  There was a brief time when it didn't print which files conflicted, which could be happening now.

Assuming that any changes you have to %home are in files which aren't in %kids, you can mount both, `cp pier/kids/* pier/home`, unmount %kids, then commit %home.  This should update you, and then the last thing you'll want to do is fixup the history with `|merge %home our %kids, =gem %this`.  `%this` is the opposite of `%that` -- it still creates a new merge commit, but it takes no changes from the source desk.  This way future updates will recognize that the conflicts have been resolved.

**Comment by botter-nidnul** [comment]
Thanks, manually merged in urbit-os-v1.0.28 to home on the unix side of things and then fixed the merge history with `|merge %home our %kids, =gem %this`. Hopefully OTAs will proceed normally now.

Still find it disturbing that `|commit`ing nothing more than a publish notebook onto my pristine boot pill home hash caused it to be unable to automerge, and even worse that the failure was mostly silent, with +trouble returning updated base hashes like OTAs were merging normally.

Going to close this issue, since this bug was fixed in urbit-os-v1.0.26, but there seems to be another bug(s) with merges and +trouble that allowed me to experience this one. I'll try to see if I can reproduce what I experienced, and will open a new issue if I succeed.
